app/src/main/java/org/simple/clinic/search/results/PhoneNumberObfuscator.kt

```diff
@@ -1,20 +1,30 @@
 package org.simple.clinic.search.results
 
+import org.simple.clinic.util.Unicode.bullet
+import org.simple.clinic.util.Unicode.hairSpace
 import javax.inject.Inject
 
 interface PhoneNumberObfuscator {
   fun obfuscate(number: String): String
 }
 
 class IndianPhoneNumberObfuscator @Inject constructor() : PhoneNumberObfuscator {
+
+  private val mask = "$bullet$hairSpace"
+
   override fun obfuscate(number: String): String {
-    if (number.length < 10) {
-      return "• ".repeat(number.length)
-    }
+    val lastThreeIndices = (number.lastIndex - 2)..number.lastIndex
+    val fiveBeforeLastThree = (lastThreeIndices.first - 5)..(lastThreeIndices.first - 1)
 
-    val first2Digits = number.substring(0, 2)
-    val last3Digits = number.substring(number.length - 3, number.length)
-    val obfuscatedDigits = "• ".repeat(number.length - 5)
-    return "$first2Digits $obfuscatedDigits$last3Digits"
+    return number
+        .mapIndexed { index, char ->
+          when (index) {
+            in lastThreeIndices -> char.toString()
+            in fiveBeforeLastThree -> mask
+            fiveBeforeLastThree.start - 1 -> "$char$hairSpace"
```

Comment:
It'll be super nice to also name this `fiveBeforeLastThree.start - 1` just like how we've named as `fiveBeforeLastThree`.